packages/xo-web/src/xo-app/sr/tab-general.js

```diff
@@ -1,25 +1,161 @@
+          return vbd && vms[vbd.VM]
+        }),
+    },
+  }),
+  injectState,
+  ({
+    state: { vmsByVdi },
+    group: { baseCopies, id, name_label: name, usage, vdis, snapshots },
+  }) => {
+    const showBaseCopies = baseCopies !== undefined && baseCopies.n > 0
+    const showVdis = showBaseCopies || snapshots.length > 0
+    let disk
+    let vm
+    return (
+      <div>
+        {vdis !== undefined ? (
+          <ul style={ulStyles}>
+            {showBaseCopies && (
+              <li>
+                {_('baseCopyTooltip', {
+                  n: baseCopies.n,
+                  usage: formatSize(baseCopies.usage),
+                })}
+              </li>
+            )}
+            {snapshots.length > 0 && (
+              <li>
+                {_('snapshotsTooltip', {
+                  n: snapshots.length,
+                  usage: formatSize(sumBy(snapshots, 'usage')),
+                })}
+              </li>
+            )}
+            {snapshots.length > 0 && (
+              <ul style={styles}>
+                {snapshots.map(({ id, name_label: name, usage }) => (
+                  <li key={id}>
+                    {_('diskTooltip', {
+                      name,
+                      usage: formatSize(usage),
+                    })}
+                  </li>
+                ))}
+              </ul>
+            )}
+            {showVdis && (
+              <li>
+                {_('vdisTooltip', {
+                  n: vdis.length,
+                  usage: formatSize(sumBy(vdis, 'usage')),
+                })}
+              </li>
+            )}
+            {showVdis ? (
+              <ul style={styles}>
+                {vdis.map(({ id, name_label: name, usage }) => (
+                  <li key={id}>
+                    {(vm = vmsByVdi[id]) === undefined
+                      ? _('diskTooltip', {
+                          name,
+                          usage: formatSize(usage),
+                        })
+                      : _('vdiOnVmTooltip', {
+                          name,
+                          usage: formatSize(usage),
+                          vmName: vm.name_label,
+                        })}
+                  </li>
+                ))}
+              </ul>
+            ) : (
+              <span>
+                {((disk = vdis[0]), (vm = vmsByVdi[disk.id])) === undefined
+                  ? _('diskTooltip', {
+                      name: disk.name_label,
+                      usage: formatSize(disk.usage),
+                    })
+                  : _('vdiOnVmTooltip', {
+                      name: disk.name_label,
+                      usage: formatSize(disk.usage),
+                      vmName: vm.name_label,
+                    })}
+              </span>
+            )}
+          </ul>
+        ) : (
+          <span>
+            {_('diskTooltip', {
+              name,
+              usage: formatSize(usage),
+            })}
+          </span>
+        )}
+      </div>
```

Comment:
Can you explain all the nested conditions (ie how the component behaves)? We'll see if we can simplify it otherwise we'll need comments.

Reply:
1.  To differentiate a group (snapshots, vdis, base copies) from orphaned snapshots, I differentiate them by the `vdis` property.
2.  for list:
       2.1.  `showBaseCopies`:  to display the count of base copies if exist
       2.2.  `showSnapshots`: 
               *  to display the  title  (n snapshots (size)
               *  to display list of snapshots  (nameSnapshot (size)
      2.3.  `showVdis`:   to differentiate  if group contain snapshots or base copies.
             * `showVdis = true`:  display title `n VDIs (size)`  and the list of VDIs
             * `showVdis = false`:  display:  `VDI name (size)`.